### PR TITLE
Added wrapper functions for STRING_AGG() function to fix issue with input containing multi-byte characters.

### PR DESCRIPTION
**Pull request by rohit01010**
### Description
Currently for string function `STRING_AGG()` we are directly using corresponding PG function whose input and output datatype is `TEXT`. This leads to incorrect output for input containing multi-byte characters with `nchar` and `nvarchar` datatype. This PR will fix this issue by adding wrapper functions with correct input and return datatype for `STRING_AGG()` function.

Cherry-picked from: https://github.com/babelfish-for-postgresql/babelfish_extensions/pull/2869

Authored-by: Rohit Bhagat [rohitbgt@amazon.com](mailto:rohitbgt@amazon.com)
Signed-off-by: Rohit Bhagat [rohitbgt@amazon.com](mailto:rohitbgt@amazon.com)

### Issues Resolved

BABEL-5156

### Test Scenarios Covered ###
* **Use case based -** YES

* **Boundary conditions -** YES

* **Arbitrary inputs -** YES

* **Negative test cases -** YES

* **Minor version upgrade tests -** YES

* **Major version upgrade tests -** YES

* **Performance tests -** NO

* **Tooling impact -** NO

* **Client tests -** NO

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: contrib/babelfishpg_tsql/src/tsqlIface.cpp [hunk 1]
```diff
@@ -8340,6 +8340,12 @@ rewrite_string_agg_query(TSqlParser::STRING_AGGContext *ctx)
 		rewritten_query_fragment.emplace(std::make_pair(ctx->order_by_clause()->start->getStartIndex(), std::make_pair(::getFullText(ctx->order_by_clause()), "")));
 		rewritten_query_fragment.emplace(std::make_pair(ctx->RR_BRACKET()[1]->getSymbol()->getStartIndex(), std::make_pair(::getFullText(ctx->RR_BRACKET()[1]), "")));
 	}
+	
+	if (ctx->STRING_AGG())
+	{
+		size_t startPosition = ctx->STRING_AGG()->getSymbol()->getStartIndex();
+		rewritten_query_fragment.emplace(std::make_pair(startPosition, std::make_pair("", "sys.")));
+	}
```
Comment (Anikait143) [inline contrib/babelfishpg_tsql/src/tsqlIface.cpp:8348]: Please add a comment specifying the need for this change

**Review by Anikait143 — APPROVED** [review]

File: contrib/babelfishpg_tsql/src/tsqlIface.cpp (hunk 1 above)
Comment (rohit01010) [reply contrib/babelfishpg_tsql/src/tsqlIface.cpp:8348]: Updated the comment above this function.

File: test/JDBC/jdbc_schedule [hunk 2]
```diff
@@ -368,3 +368,10 @@ ignore#!#BABEL_4330-before-15_8-or-16_4-vu-verify
 ignore#!#alter-procedure-before-15_8-or-16_4-vu-prepare
 ignore#!#alter-procedure-before-15_8-or-16_4-vu-verify
 ignore#!#alter-procedure-before-15_8-or-16_4-vu-cleanup
+
+ignore#!#string_agg-before-15_9-or-16_5-vu-prepare
+ignore#!#string_agg-before-15_9-or-16_5-vu-verify
+ignore#!#string_agg-before-15_9-or-16_5-vu-cleanup
+ignore#!#string_agg-before-14_5-vu-prepare
+ignore#!#string_agg-before-14_5-vu-verify
+ignore#!#string_agg-before-14_5-vu-cleanup
```
Comment (Anikait143) [inline test/JDBC/jdbc_schedule:377]: Please ignore `string_agg_within-16_4-vu-*` files as well

File: test/JDBC/upgrade/16_4/schedule [hunk 3]
```diff
@@ -556,7 +556,7 @@ BABEL-3820
 stuff
 PARTITION
 BABEL-5071
-string_agg
+string_agg_within
```
Comment (Anikait143) [inline test/JDBC/upgrade/16_4/schedule]: we should add `string_agg-before-15_9-or-16_5` and  `string_agg_within-16_4` instead of `string_agg_within`

**Review by Anikait143 — COMMENTED** [review]
Also add test files to other remaining upgrade schedule files

File: test/JDBC/upgrade/16_4/schedule (hunk 3 above)
Comment (rohit01010) [reply test/JDBC/upgrade/16_4/schedule]: done

File: test/JDBC/jdbc_schedule (hunk 2 above)
Comment (rohit01010) [reply test/JDBC/jdbc_schedule:377]: done

**Review by Anikait143 — APPROVED** [review]

**Review by Deepesh125 — APPROVED** [review]

**Pull request merged**